waiter/config-full.edn

```diff
@@ -350,6 +350,11 @@
                                 "scale-factor" 1
                                 "scale-up-factor" 0.1}
 
+ ;; Service description parameters with numeric values, e.g. cpus, can be configured
+ ;; to have maximum value limits.
+ :service-description-upper-limits {"cpus" 40
```

Comment:
It's nice to be general but numeric limits only apply to specific parameters.  We can't have a numeric limit on `health-check-url` for example.

I'd prefer `:service-constraints {:max-cpus 40 :max-mem 32768}` and handle these specifically.  Seems simple to me.  It feels awkward to use `schema` to enforce these configurable limits.

And if you're not a fan of that, then maybe: 

`:service-description-constraints {"cpu" {:max 40} "mem" {:max 32768}}` which leaves the door open to non-numeric limits, e.g. {"backend-proto" {:is "https"}}

Reply:
I ended up using `:service-description-constraints {"cpu" {:max 40} "mem" {:max 32768}}` and `schema`.